databases/week3/Meal_query.sql

```diff
@@ -0,0 +1,385 @@
+select * from reservation where id = 4;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+update reservation
+set
+    number_of_guests = 8,
+    meal_id = 1
+where
+    id = 4;
+
+--Delete a reservation with any id, fx 1
+delete from reservation where id = 4;
+
+--Queries for review
+-- Get all reviews
+select * from review;
+
+--Add a new review
+insert into
+    review
+values (
+        4,
+        'nasty tagine',
+        'Worst tagine I ever tasted',
+        5,
+        1,
+        '2024-06-05'
+    );
+
+--Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+update review
+set
+    title = 'good',
+    description = 'tastes good'
+where
+    id = 4;
+
+-- Delete a review with any id, fx 1
+delete from review where id = 4;
+
+--Additional queries
+insert into
+    meal
+VALUES (
+        9,
+        'Burger',
+        ' It is made up of bun with vegtable patty and tomato sauce',
+        'USA',
+        '2024-07-18 20:00:03',
+        50,
+        20,
+        '2024-07-11'
+    ),
+    (
+        10,
+        'Chenguan',
+        ' It is made up of pasta with chicken patty and chilli sauce',
+        'Finland',
+        '2024-07-19 20:00:03',
+        30,
+        10,
+        '2024-07-11'
+    );
+
+insert into
+    reservation
+values (
+        4,
+        3,
+        9,
+        '2024-07-12',
+        6589745,
+        'Kirm Johnson',
+        'kirm@gmail.com'
+    ),
+    (
+        5,
+        2,
+        10,
+        '2024-07-12',
+        7896542,
+        'Vizhurathi',
+        'viz@example.com'
+    );
+
+insert into
+    review
+values (
+        4,
+        'Bad Burger',
+        'Very bad tasty burger',
+        9,
+        1,
+        '2024-07-12'
+    ),
+    (
+        5,
+        'Good',
+        'Very tasty dish',
+        10,
+        5,
+        '2024-07-12'
+    );
+
+-- Get meals that has a price smaller than a specific price fx 90
+select * from meal where price < 20;
+
+-- Get meals that still has available reservations
+select *
+from meal
+    left join (
+        select meal_id, sum(number_of_guests) as guests
+        from reservation
+        group by
+            meal_id
+    ) as A on meal.id = A.meal_id
+where
+    meal.max_reservations > COALESCE(A.guests, 0);
+
+--Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+select * from meal where title like '%pasta%';
+
+-- Get meals that has been created between two dates
+select *
+from meal
+where
+    created_date BETWEEN '2024-07-09' and '2024-07-12';
+
+-- Get only specific number of meals fx return only 5 meals
+select * from meal order by rand() limit 5;
+
+-- Get the meals that have good reviews
+select meal.id, meal.title, rev.title, rev.stars, rev.meal_id
+from meal
+    join (
+        select *
+        from review
+        where
+            stars > 3
+    ) as rev on meal.id = rev.meal_id;
```

Comment:
Its good that you practice inner queries, but I think a simple join was enough here.

Reply:
Yes, I was just complicate the things . I could use normal join